src/DependencyInjection/Configuration.php

```diff
@@ -78,7 +78,7 @@ private function addMetadataSection(ArrayNodeDefinition $node): void
                     ->fixXmlConfig('directory', 'directories')
                     ->children()
                         ->scalarNode('cache')->defaultValue('file')->end()
-                        ->scalarNode('type')->defaultValue('annotation')->end()
+                        ->scalarNode('type')->defaultValue(PHP_MAJOR_VERSION > 7 ? 'attribute' : 'annotation')->end()
```

Comment:
This is a BC break for php8 user which didn't migrated from annotation.
(I am one of them).

I do appreciate the idea, but it require a major release...

Reply:
It is indeed. It's noted in the CHANGELOG

Reply:
People might miss the changelog when bumping dependencies if their composer use the ^ constraint. That's why it require a new major version Imho